tests/Fixer/Casing/MagicMethodCasingFixerTest.php

```diff
@@ -0,0 +1,163 @@
+        $methodNames = ['__call', '__set'];
+
+        foreach ($methodNames as $i => $name) {
+            unset($allMethodNames[$name]);
+
+            yield sprintf('method declaration for "%s".', $name) => [
+                sprintf('<?php class Foo {public function %s($a, $b){}}', $name),
+                sprintf('<?php class Foo {public function %s($a, $b){}}', strtoupper($name)),
+            ];
+        }
+
+        foreach ($methodNames as $i => $name) {
+            yield sprintf('method call "%s".', $name) => [
+                sprintf('<?php $a->%s($a, $b);', $name),
+                sprintf('<?php $a->%s($a, $b);', strtoupper($name)),
+            ];
+        }
+
+        // single argument
+        $methodNames = ['__get', '__isset', '__unset'];
+
+        foreach ($methodNames as $i => $name) {
+            unset($allMethodNames[$name]);
+
+            yield sprintf('method declaration for "%s".', $name) => [
+                sprintf('<?php class Foo {public function %s($a){}}', $name),
+                sprintf('<?php class Foo {public function %s($a){}}', strtoupper($name)),
+            ];
+        }
+
+        foreach ($methodNames as $i => $name) {
+            yield sprintf('method call "%s".', $name) => [
+                sprintf('<?php $a->%s($a);', $name),
+                sprintf('<?php $a->%s($a);', strtoupper($name)),
+            ];
+        }
+
+        // no argument
+
+        foreach ($allMethodNames as $i => $name) {
+            yield sprintf('method declaration for "%s".', $name) => [
+                sprintf('<?php class Foo {public function %s(){}}', $name),
+                sprintf('<?php class Foo {public function %s(){}}', strtoupper($name)),
+            ];
+        }
+
+        foreach ($allMethodNames as $i => $name) {
+            yield sprintf('method call "%s".', $name) => [
+                sprintf('<?php $a->%s();', $name),
+                sprintf('<?php $a->%s();', strtoupper($name)),
+            ];
+        }
+    }
+
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideDoNotFixCases
+     */
+    public function testDoNotFix($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideDoNotFixCases()
+    {
+        return [
+            [
+                '<?php
+                    $a->a();
+                    $a->__not_magic();
+
+                    function a()
+                    {
+                    }
+
+                    function __()
+                    {
+                    }
+
+                    function __alsoNotMagic()
+                    {
+                    }
+
+                    class B
+                    {
+                        public function _not_magic()
+                        {
+                        }
+                    }
+
+                    #->__sleep()
+
+                    echo $a->__sleep;
+                ',
+            ],
```

Comment:
Please add the following cases:
```php
<?php
function __Tostring() {}
```
```php
<?php
__Tostring();
```

Reply:
```
<?php
function __Tostring() {}
```

can we note this as a `false`-fixing and move on, like anyone who creates function with magic prefix `__` is not likely to use this tool anyway ;)

Reply:
the one who will inherit codebase after him would ;P

agree, edge case...